cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PullCommand.java

```diff
@@ -0,0 +1,159 @@
+    private PrintStream errStream;
+    private CentralAPIClient client;
+
+    @CommandLine.Parameters private List<String> argList;
+
+    @CommandLine.Option(names = { "--help", "-h" }, hidden = true) private boolean helpFlag;
+
+    @CommandLine.Option(names = "--debug", hidden = true) private String debugPort;
+
+    public PullCommand() {
+        this.errStream = System.err;
+        this.client = new CentralAPIClient();
+    }
+
+    public PullCommand(PrintStream errStream, CentralAPIClient client) {
+        this.errStream = errStream;
+        this.client = client;
+    }
+
+    @Override
+    public void execute() {
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(PULL_COMMAND);
+            errStream.println(commandUsageInfo);
+            return;
+        }
+
+        if (argList == null || argList.isEmpty()) {
+            throw LauncherUtils.createUsageExceptionWithHelp("no package given");
+        }
+
+        if (argList.size() > 1) {
+            throw LauncherUtils.createUsageExceptionWithHelp("too many arguments");
+        }
+
+        // Enable remote debugging
+        if (null != debugPort) {
+            System.setProperty(SYSTEM_PROP_BAL_DEBUG, debugPort);
+        }
+
+        String resourceName = argList.get(0);
+        String orgName;
+        String packageName;
+        String version;
+
+        if (!validPackageName(resourceName)) {
+            CommandUtil.printError(errStream, "invalid package name. Provide the package name with the org name ",
+                    "ballerina pull {<org-name>/<package-name> | <org-name>/<package-name>:<version>}", false);
+            Runtime.getRuntime().exit(1);
+            return;
+        }
+
+        // Get org name
+        String[] moduleInfo = resourceName.split("/");
+        orgName = moduleInfo[0];
+        String moduleNameAndVersion = moduleInfo[1];
+
+        // Get package name
+        String[] packageInfo = moduleNameAndVersion.split(":");
+        if (packageInfo.length == 2) {
+            packageName = packageInfo[0];
+            version = packageInfo[1];
+        } else {
+            packageName = moduleNameAndVersion;
+            version = Names.EMPTY.getValue();
+        }
+
+        Path packagePathInBaloCache = RepoUtils.createAndGetHomeReposPath()
+                .resolve(ProjectConstants.BALO_CACHE_DIR_NAME).resolve(orgName).resolve(packageName);
+        // create directory path in balo cache
+        try {
+            createDirectory(packagePathInBaloCache);
+        } catch (IOException e) {
+            throw createLauncherException(
+                    "unexpected error occurred while creating package repository in balo cache: " + e.getMessage());
+        }
+
+        for (String supportedPlatform : SUPPORTED_PLATFORMS) {
+            this.client.pullPackage(orgName, packageName, version, packagePathInBaloCache, supportedPlatform, false);
+        }
+    }
```

Comment:
Should `Runtime.getRuntime().exit(0);` come here ?

Reply:
Added